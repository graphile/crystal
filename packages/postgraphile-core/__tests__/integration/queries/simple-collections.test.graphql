query {
  a: allPeopleList { ...personFragment }
  b: allPeopleList(first: 2) { ...personFragment }
	c: allPeopleList(orderBy: NAME_ASC) { ...personFragment }
  d: allPeopleList(orderBy: NAME_DESC) { ...personFragment }
  e: allPostsList(condition: { authorId: 2 }) { ...postFragment }
	f: allPostsList(first: 2, condition: { authorId: 2 }) { ...postFragment }
  g: allPeopleList(first: 3, offset: 1) { ...personFragment }
  h: allPeopleList(first: 0) { ...personFragment }
  i: allPeopleList(orderBy: PRIMARY_KEY_ASC) { ...personFragment }
  j: allPeopleList(condition: { about: null }) { ...personFragment }
  k: allPostsList(orderBy: [AUTHOR_ID_DESC, HEADLINE_DESC], first: 3) { ...postFragment }

}

fragment personFragment on Person {
  id
  name
  email
}

fragment postFragment on Post {
  headline
  authorId
}
