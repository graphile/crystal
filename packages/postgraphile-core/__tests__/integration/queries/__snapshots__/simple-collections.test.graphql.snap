// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`execute 1`] = `
Object {
  "data": Object {
    "a": Array [
      Object {
        "email": "john.smith@email.com",
        "id": 1,
        "name": "John Smith",
      },
      Object {
        "email": "sara.smith@email.com",
        "id": 2,
        "name": "Sara Smith",
      },
      Object {
        "email": "budd.deey@email.com",
        "id": 3,
        "name": "Budd Deey",
      },
      Object {
        "email": "kathryn.ramirez@email.com",
        "id": 4,
        "name": "Kathryn Ramirez",
      },
      Object {
        "email": "joe.tucker@email.com",
        "id": 5,
        "name": "Joe Tucker",
      },
      Object {
        "email": "graphile-build.issue.27.exists@example.com",
        "id": 6,
        "name": "Twenty Seventwo",
      },
    ],
    "b": Array [
      Object {
        "email": "john.smith@email.com",
        "id": 1,
        "name": "John Smith",
      },
      Object {
        "email": "sara.smith@email.com",
        "id": 2,
        "name": "Sara Smith",
      },
    ],
    "c": Array [
      Object {
        "email": "budd.deey@email.com",
        "id": 3,
        "name": "Budd Deey",
      },
      Object {
        "email": "joe.tucker@email.com",
        "id": 5,
        "name": "Joe Tucker",
      },
      Object {
        "email": "john.smith@email.com",
        "id": 1,
        "name": "John Smith",
      },
      Object {
        "email": "kathryn.ramirez@email.com",
        "id": 4,
        "name": "Kathryn Ramirez",
      },
      Object {
        "email": "sara.smith@email.com",
        "id": 2,
        "name": "Sara Smith",
      },
      Object {
        "email": "graphile-build.issue.27.exists@example.com",
        "id": 6,
        "name": "Twenty Seventwo",
      },
    ],
    "d": Array [
      Object {
        "email": "graphile-build.issue.27.exists@example.com",
        "id": 6,
        "name": "Twenty Seventwo",
      },
      Object {
        "email": "sara.smith@email.com",
        "id": 2,
        "name": "Sara Smith",
      },
      Object {
        "email": "kathryn.ramirez@email.com",
        "id": 4,
        "name": "Kathryn Ramirez",
      },
      Object {
        "email": "john.smith@email.com",
        "id": 1,
        "name": "John Smith",
      },
      Object {
        "email": "joe.tucker@email.com",
        "id": 5,
        "name": "Joe Tucker",
      },
      Object {
        "email": "budd.deey@email.com",
        "id": 3,
        "name": "Budd Deey",
      },
    ],
    "e": Array [
      Object {
        "authorId": 2,
        "headline": "No… It’s a thing; it’s like a plan, but with more greatness.",
      },
      Object {
        "authorId": 2,
        "headline": "It’s a fez. I wear a fez now. Fezes are cool.",
      },
      Object {
        "authorId": 2,
        "headline": "What’s with you kids? Every other day it’s food, food, food.",
      },
    ],
    "f": Array [
      Object {
        "authorId": 2,
        "headline": "No… It’s a thing; it’s like a plan, but with more greatness.",
      },
      Object {
        "authorId": 2,
        "headline": "It’s a fez. I wear a fez now. Fezes are cool.",
      },
    ],
    "g": Array [
      Object {
        "email": "sara.smith@email.com",
        "id": 2,
        "name": "Sara Smith",
      },
      Object {
        "email": "budd.deey@email.com",
        "id": 3,
        "name": "Budd Deey",
      },
      Object {
        "email": "kathryn.ramirez@email.com",
        "id": 4,
        "name": "Kathryn Ramirez",
      },
    ],
    "h": Array [],
    "i": Array [
      Object {
        "email": "john.smith@email.com",
        "id": 1,
        "name": "John Smith",
      },
      Object {
        "email": "sara.smith@email.com",
        "id": 2,
        "name": "Sara Smith",
      },
      Object {
        "email": "budd.deey@email.com",
        "id": 3,
        "name": "Budd Deey",
      },
      Object {
        "email": "kathryn.ramirez@email.com",
        "id": 4,
        "name": "Kathryn Ramirez",
      },
      Object {
        "email": "joe.tucker@email.com",
        "id": 5,
        "name": "Joe Tucker",
      },
      Object {
        "email": "graphile-build.issue.27.exists@example.com",
        "id": 6,
        "name": "Twenty Seventwo",
      },
    ],
    "j": Array [
      Object {
        "email": "john.smith@email.com",
        "id": 1,
        "name": "John Smith",
      },
      Object {
        "email": "sara.smith@email.com",
        "id": 2,
        "name": "Sara Smith",
      },
      Object {
        "email": "kathryn.ramirez@email.com",
        "id": 4,
        "name": "Kathryn Ramirez",
      },
      Object {
        "email": "joe.tucker@email.com",
        "id": 5,
        "name": "Joe Tucker",
      },
      Object {
        "email": "graphile-build.issue.27.exists@example.com",
        "id": 6,
        "name": "Twenty Seventwo",
      },
    ],
    "k": Array [
      Object {
        "authorId": 5,
        "headline": "Please, Don-Bot… look into your hard drive, and open your mercy file!",
      },
      Object {
        "authorId": 3,
        "headline": "You know how I sometimes have really brilliant ideas?",
      },
      Object {
        "authorId": 3,
        "headline": "They’re not aliens, they’re Earth…liens!",
      },
    ],
  },
}
`;
